graphql-api/src/queries/variant-datasets/gnomad-v4-variant-queries.ts

```diff
@@ -466,6 +478,20 @@ const fetchVariantsByGene = async (esClient: any, gene: any, subset: Subset) =>
       )
       .map(shapeVariantSummary(subset, { type: 'gene', geneId: gene.gene_id }))
 
+    const lofCurationResults = await fetchLofCurationResultsByGene(esClient, 'v4', gene)
```

Comment:
I'm assuming that you're following certain patterns laid down by the legacy code that already exists, but there are cleaner ways to write all this: we don't have to follow the existing code just because it exists.

Off the top of my head I'd do something like:

```
const keyedLofCurationResults = lofCurationResultsByVariant.map((result) => [result.gene_id, result])
const lofCurationResultsByVariant = Object.fromEntries(keyedLofCurationResults)

return shapedHits.map((hit) => ({...hit, lof_curation: lofCurationResultsByVariant[variant.variant_id]})
```